src/components/uncompress.js

```diff
@@ -79,12 +79,14 @@ class Uncompress extends Component {
   }
 
   extract = async blob => {
+    const { preview } = this.context.state
     try {
       // Compressed files 1437
       const list = await this.openArchive(blob)
 
       if (list && list.length > 0) {
-        await asyncForEach(list, async (item, index) => {
+        const items = preview ? list.splice(0,3) : list
```

Comment:
The limit should be configurable, also probably a `prop`

Reply:
You prefer 2 props like preview & previewCount or only one that active preview if value is greater than 1?